src/devices/wifi/nm-iwd-manager.c

```diff
@@ -648,9 +649,14 @@ static void
 nm_iwd_manager_init (NMIwdManager *self)
 {
 	NMIwdManagerPrivate *priv = NM_IWD_MANAGER_GET_PRIVATE (self);
+	NMManager *manager = nm_manager_get ();
 
 	priv->cancellable = g_cancellable_new ();
+
 	prepare_object_manager (self);
+
+	g_signal_connect (manager, "device-added",
+	                  G_CALLBACK (device_added), self);
```

Comment:
The later commit "iwd: disconnect from NMManager signals in cleanup" should be squashed into this one.

When you connect to a signal, you commonly have to disconnect too. If you are going to disconnect later, you must be sure that the object is still alive. `nm_manager_get()` is a singleton, but during shutdown singletons are deconstructed (well, currently we leak a lot of singletons, that is a bug). If a singleton uses another singleton just for the moment, it might not need to take a reference. But if you connect and disconnect to a signal that implies that the source singleton needs to stay alive during that time.

I guess, you could use `g_signal_connect_object()`, or maybe better: add a `NMIwdManagerPrivate.manager` field that keeps a reference to the manager instance. Initialize it in `nm_iwd_manager_init()`, and use it instead of `nm_manager_get()`.

Reply:
Good point, let's keep a reference to NMManager. 
I think this is a fix unrelated from the gdbus workaround patch so let me just reorder the patches instead of squashing them.